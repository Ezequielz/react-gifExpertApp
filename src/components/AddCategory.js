import React, { useState } from 'react'
import PropTypes from 'prop-types'


export const AddCategory = ({setCategories}) => {


    const [inpuntValue, setinpuntValue] = useState("");


    const handleInpuntChange = (e) =>{
        setinpuntValue( e.target.value)
    } 
    
    const handleSubmit = (e) => {
        e.preventDefault()

        if( inpuntValue.trim().length > 2){
            setCategories( cats => [inpuntValue,...cats])
            setinpuntValue("")
        }
        
       
    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value = {inpuntValue}
                onChange={ handleInpuntChange}
            />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}