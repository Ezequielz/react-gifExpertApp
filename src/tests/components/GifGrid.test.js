import React from 'react';
import "@testing-library/jest-dom"

import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from '../../hooks/useFetchGifs';
import { getGifs } from '../../helpers/getGifs';

jest.mock("../../hooks/useFetchGifs")

describe('Pruebas en <GifGrid />', () => {
    
    const category = "Hola mundo"
    
    test('debe mostrarse correctamente (snapshot) ', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow ( <GifGrid category= { category }/>)
        
        expect( wrapper ).toMatchSnapshot()
        
    })
    
    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
       
        const gifs = [{
            id: "ABC",
            url: "https://localhost",
            title: "cualquiier cosa"
        },
        {
            id: "123",
            url: "https://localhost2",
            title: "cualquiier cosa 2"
        }];

        useFetchGifs.mockReturnValue({
            data: gifs, 
            loading: false
        })

        const wrapper = shallow ( <GifGrid category= { category }/>)

        expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find("p").exists() ).toBe(false)
        expect( wrapper.find("GifGridItem").length ).toBe( gifs.length )

        })
        

})
