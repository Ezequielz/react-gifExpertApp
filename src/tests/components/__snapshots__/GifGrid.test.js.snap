// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="cat animate__animated animate__fadeIn"
  >
     
    Hola mundo
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquiier cosa"
      url="https://localhost"
    />
    <GifGridItem
      id="123"
      key="123"
      title="cualquiier cosa 2"
      url="https://localhost2"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe mostrarse correctamente (snapshot)  1`] = `
<Fragment>
  <h3
    className="cat animate__animated animate__fadeIn"
  >
     
    Hola mundo
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
